readers/cassandra/messages.go

```diff
@@ -75,3 +73,31 @@ func (cr cassandraRepository) ReadAll(chanID string, offset, limit uint64) []mai
 
 	return page
 }
+
+func buildQuery(chanID string, offset, limit uint64, andQuery map[string]string) (string, []interface{}) {
+	var condSql string
+	var values []interface{}
+
+	cql := `SELECT channel, subtopic, publisher, protocol, name, unit,
+			value, string_value, bool_value, data_value, value_sum, time,
+			update_time, link FROM messages WHERE channel = ? %s LIMIT ?
+			ALLOW FILTERING`
+
+	values = append(values, chanID)
+
+	for name, value := range andQuery {
+		switch name {
+		case
+			"channel",
+			"subtopic",
+			"publisher",
+			"name",
+			"protocol":
+			condSql += fmt.Sprintf(` AND %s = ?`, name)
```

Comment:
You can user `fmt.Sprintf` to append: `condSql = fmt.Sprintf("%s AND %s = ?", condSql, name)`.

Reply:
Yes, @dusanb94 's proposition is more readable IMHO